pkg/scheduler/queue/tenant_queues_test.go

```diff
@@ -19,8 +19,47 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func (qb *queueBroker) enqueueObjectsForTests(tenantID TenantID, numObjects int) error {
+	for i := 0; i < numObjects; i++ {
+		err := qb.tree.EnqueueBackByPath(QueuePath{string(tenantID)}, &tenantRequest{
+			tenantID: tenantID,
+			req:      fmt.Sprintf("%v: object-%v", tenantID, i),
+		})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func buildExpectedObject(tenantID TenantID, num int) *tenantRequest {
+	return &tenantRequest{
+		tenantID: tenantID,
+		req:      fmt.Sprintf("%v: object-%v", tenantID, num),
+	}
+}
+
+type dequeueVal struct {
+	req    *tenantRequest
+	tenant *queueTenant
+}
+
+func assertExpectedValuesOnDequeue(t *testing.T, qb *queueBroker, lastTenantIndex int, querierID QuerierID, expectedVals []dequeueVal) int {
+	var req *tenantRequest
+	var tenant *queueTenant
+	var err error
+
+	for _, expected := range expectedVals {
+		req, tenant, lastTenantIndex, err = qb.dequeueRequestForQuerier(lastTenantIndex, querierID)
+		assert.Equal(t, expected.req, req)
+		assert.Equal(t, expected.tenant, tenant)
+		assert.NoError(t, err)
+	}
+	return lastTenantIndex
+}
+
 func TestQueues(t *testing.T) {
```

Comment:
I would love to get a bit more specific here with the test name - something like `TestQueues_UnshardedTenantRotation` or whatever you decide the test is really about